lib/Application/Complete.php

```diff
@@ -160,11 +160,15 @@ private function populateSuggestions(SymbolContext $symbolInformation, Type $typ
             return $symbolInformation->withIssue(sprintf('Could not find class "%s"', (string) $type));
         }
 
+        $needsPublicAccess = !in_array($symbolInformation->symbol()->name(), ['this', 'self']);
         /** @var $method ReflectionMethod */
         foreach ($classReflection->methods() as $method) {
```

Comment:
I think the next step (after this PR) would be to add `$classReflection->methods()->visibleTo($symbolInformation->containerType())`